types/scte224v20200407/sctestructures.go

```diff
@@ -0,0 +1,325 @@
+				if nil == convErr {
+					hours = 24 * days
+				} else {
+					log.Println(convErr)
+				}
+			}
+
+			if "" != subMatches[2] {
+				parsedDur, convErr := time.ParseDuration(strings.ToLower(subMatches[2]))
+				if nil == convErr {
+					nanoseconds = parsedDur
+				} else {
+					log.Println(convErr)
+				}
+			}
+			duration = nanoseconds + time.Duration(hours)*time.Hour
+		}
+	}
+	return duration
+}
+
+type AltID struct {
+	XMLName     xml.Name `xml:"http://www.scte.org/schemas/224 AltID" json:"-"`
+	Description string   `xml:"description,attr,omitempty" json:"description,omitempty"`
+	Value       string   `xml:",chardata" json:"value,omitempty"`
+	Type        string   `xml:"type,attr,omitempty" json:"type,omitempty"`
+}
+
+//Table 10
+type Apply struct {
+	XMLName  xml.Name `xml:"http://www.scte.org/schemas/224 Apply" json:"-"`
+	Duration Duration `xml:"duration,attr,omitempty" json:"duration,omitempty"`
+	Priority *uint    `xml:"priority,attr,omitempty" json:"priority,omitempty"`
+	Policy   *Policy  `xml:"http://www.scte.org/schemas/224 Policy,omitempty" json:"policy,omitempty"`
+}
+
+func (ap *Apply) HasExplicitPriority() bool {
+	return nil != ap.Priority
+}
+
+func (ap *Apply) GetPriority() uint {
+	if ap.HasExplicitPriority() {
+		return *ap.Priority
+	}
+	return 0
+}
+
+//Table 9
+type Remove struct {
+	XMLName xml.Name `xml:"http://www.scte.org/schemas/224 Remove" json:"-"`
+	Policy  *Policy  `xml:"http://www.scte.org/schemas/224 Policy,omitempty" json:"policy,omitempty"`
+}
+
+const matchSignalSchemaDefault = "http://www.scte.org/schemas/35"
+
+//Table 8
+type MatchSignal struct {
+	XMLName         xml.Name  `xml:"http://www.scte.org/schemas/224 MatchSignal" json:"-"`
+	Match           Match     `xml:"match,attr,omitempty" json:"match,omitempty"`
+	SignalTolerance Duration  `xml:"signalTolerance,attr,omitempty" json:"signalTolerance,omitempty"`
+	Assertions      []*Assert `xml:"http://www.scte.org/schemas/224 Assert,omitempty" json:"assertions,omitempty"`
+	Schema          string    `xml:"schema,attr,omitempty" json:"schema,omitempty"`
+}
+
+// Custom unmarshalling to provide default value to "schema" attribute, when unavailable
+func (ms *MatchSignal) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	/**
+	* https://pkg.go.dev/encoding/xml#Unmarshaler
+	* One common implementation strategy is to unmarshal into a separate value with a layout matching the expected XML using d.DecodeElement,
+	* and then to copy the data from that value into the receiver.
+	* this is to prevent recursive unmarshaling ending in an infinite loop
+	 */
+	type dupMatchSignalType MatchSignal
+	matchSignal := dupMatchSignalType{}
+
+	err := d.DecodeElement(&matchSignal, &start)
+	if err != nil {
+		return err
+	}
+
+	if strings.Trim(matchSignal.Schema, " ") == "" {
+		// set the default value
+		matchSignal.Schema = matchSignalSchemaDefault
+	}
+
+	*ms = MatchSignal(matchSignal)
+	return nil
+}
```

Comment:
A custom unmarshaller for MatchSignal so we can set default value to schema, if it doesn't exist